### PR TITLE
💄 UI improvement in [View/details] page

### PR DESCRIPTION
**Pull request by AuroraHuang22**
- correct the version by fetching data from records data instead of metadata
- show file preview as long as the content type is image
- add progress indicator for loading preview image

### CODE REVIEW

File: pages/view/_iscnId.vue [hunk 1]
```diff
@@ -467,7 +467,7 @@ export default class ViewIscnIdPage extends Vue {
   }
 
   get recordData() {
-    return this.record?.data
+    return this.record && this.record.data
```
Comment (nwingt) [inline pages/view/_iscnId.vue:470]: TypeScript not working?

File: pages/view/_iscnId.vue [hunk 2]
```diff
@@ -505,6 +505,10 @@ export default class ViewIscnIdPage extends Vue {
     return `${ISCN_RAW_DATA_ENDPOINT}${this.iscnId}`
   }
 
+  get version() {
+    return this.recordData.recordVersion
```
Comment (nwingt) [inline pages/view/_iscnId.vue:508]: `recordData` may be undefined,

File: components/MetadataCard.vue [hunk 3]
```diff
@@ -57,6 +59,7 @@ export default class MetadataCard extends Vue {
   @Prop(Object) readonly allExif!: Object
 
   exifInfo: Object = {}
+  isLoading: boolean = true
```
Comment (nwingt) [inline components/MetadataCard.vue:62]: I suggest renaming this to more specify `hasLoadedImage`, with default is `false`

**Review by nwingt — CHANGES_REQUESTED** [review]

File: pages/view/_iscnId.vue (hunk 1 above)
Comment (AuroraHuang22) [reply pages/view/_iscnId.vue]: After I re-run dev it works fine.

**Review by williamchong — APPROVED** [review]

**Pull request merged**